lib/Router.js

```diff
@@ -317,15 +318,18 @@ Router.prototype.handle = function(request, connection) {
 
         // Make sure numbers are numeric
         var props = route.schema.properties || route.schema.items.properties;
+
         if (props.hasOwnProperty(paramName)) {
-            switch (props[paramName].type) {
-                case 'integer':
+            //If null are allowed and the value is null than don't make the conversions or checks below
+            if (!(paramValue === null && props[paramName].type.indexOf('null') !== -1)) {
+                if (props[paramName].type.indexOf('integer') !== -1) {
                     paramValue = parseInt(paramValue);
-                    break;
+                }
+
 
-                case 'number':
+                if (props[paramName].type.indexOf('number') !== -1) {
```

Comment:
agreed... i'll do that refactoring after I merge this in and write the browse api endpoint because I need this merged in ASAP to finish testing the browse api i'm writing...